packages/app/src/ui/pages/PresentVerifiableCredential/usePresentVerifiableCredential.ts

```diff
@@ -0,0 +1,253 @@
+
+    if (!verifiablePresentation) {
+      return;
+    }
+
+    const address = ethWallet.address?.toLowerCase();
+    const signer = await ethWallet.provider?.getSigner();
+
+    if (!address || !signer) {
+      setError("Could not connect to Ethereum account.");
+      return;
+    }
+
+    try {
+      const serializedVerifiablePresentation = serializeVerifiablePresentation(verifiablePresentation);
+      const signature = await signer.signMessage(serializedVerifiablePresentation);
+      const signedVerifiablePresentation = {
+        ...verifiablePresentation,
+        proof: [
+          {
+            type: [ETHEREUM_SIGNATURE_SPECIFICATION_TYPE],
+            proofPurpose: VERIFIABLE_CREDENTIAL_PROOF_PURPOSE,
+            verificationMethod: address,
+            created: new Date(),
+            proofValue: signature,
+          },
+        ],
+      };
+
+      await dispatch(generateVerifiablePresentation(signedVerifiablePresentation));
+      onCloseModal();
+    } catch (e) {
+      setError("Failed to sign Verifiable Presentation.");
+    }
+  }, [
+    ethWallet,
+    setError,
+    dispatch,
+    onCloseModal,
+    generateVerifiablePresentation,
+    createVerifiablePresentationFromSelectedCredentials,
+  ]);
+
+  const onSubmitVerifiablePresentationWithCryptkeeper = useCallback(async () => {
+    const verifiablePresentation = createVerifiablePresentationFromSelectedCredentials();
+
+    if (!verifiablePresentation) {
+      return;
+    }
+
+    const address = cryptKeeperWallet.address?.toLowerCase();
+
+    if (!address) {
+      setError("Could not connect to Cryptkeeper account.");
+      return;
+    }
+
+    await dispatch(generateVerifiablePresentationWithCryptkeeper({ verifiablePresentation, address }));
+    onCloseModal();
+  }, [
+    cryptKeeperWallet,
+    setError,
+    dispatch,
+    onCloseModal,
+    generateVerifiablePresentationWithCryptkeeper,
+    createVerifiablePresentationFromSelectedCredentials,
+  ]);
+
+  const onSubmitVerifiablePresentationWithoutSignature = useCallback(async () => {
+    const verifiablePresentation = createVerifiablePresentationFromSelectedCredentials();
+
+    if (!verifiablePresentation) {
+      return;
+    }
+
+    await dispatch(generateVerifiablePresentation(verifiablePresentation));
+    onCloseModal();
+  }, [
+    setError,
+    dispatch,
+    onCloseModal,
+    generateVerifiablePresentation,
+    createVerifiablePresentationFromSelectedCredentials,
+  ]);
+
+  // eslint-disable-next-line consistent-return
+  const onSubmitVerifiablePresentation = useCallback(async () => {
+    switch (true) {
+      case menuSelectedIndex === 0 && !isWalletConnected:
+        return onConnectWallet();
+      case menuSelectedIndex === 0 && isWalletConnected:
+        return onSubmitVerifiablePresentationWithMetamask();
+      case menuSelectedIndex === 1:
+        return onSubmitVerifiablePresentationWithCryptkeeper();
+      case menuSelectedIndex === 2:
```

Comment:
These numbers are magic ones. Maybe it's possible to use enum here?

Reply:
fixed